### ISSUE TITLE
Defining enum props without exporting an enum object

### ISSUE THREAD

**Issue by bighuggies** [uber/react-view#6]
First off, the library is really nice 👍 

I'm trying it out and I'm wondering if there is a way to define knobs for props as enums without using the enum object pattern?

e.g. instead of using `SIZE.default` as in the [button example](https://github.com/uber/react-view/blob/master/examples/basic.tsx#L32), just use the string `'default'`?

We use string literal unions in TypeScript for enum props so we don't export enum objects, we instead rely on the compiler to check the correct strings are being supplied.

It doesn't look like it's currently possible in [knob.tsx](https://github.com/uber/react-view/blob/master/src/ui/knob.tsx#L163). If no `enumName` is provided it uses the prop name as the name of the enum.

I know it's possible to get around this by using `useView` and creating a custom `Knobs` component - but it would be nice to re-use `Knobs` given how much it does out of the box.

**Comment by tajo** [comment]
So you just always do

```js
<Button size="default" />
``` 

and have no export of constants?

**Comment by bighuggies** [comment]
Correct, the compiler verifies the prop statically at build time.

There are also potentially some cases where the enum value is a number or some other type.

**Comment by tajo** [comment]
We could add a new flag for enum props, something like `noImport: true`? Probably makes more sense than creating a new proptype / knob.

Would you like to open a PR?

**Comment by bighuggies** [comment]
Sure, I can take a look at it.

**Comment by tajo** [comment]
Thanks! It should not be that big of a change but let me know if you need any help.

**Comment by EvanLovely** [comment]
Would love this as well! Seems to me the change would be around here:

https://github.com/uber/react-view/blob/d5aa73d396c86455f855d7458ca37c0e82577937/src/ui/knob.tsx#L213-L221

The `opt` in there is from `Object.keys(options)` (which would need to be switched to `Object.values(options)` and have the ability to not have it appended onto the `enumName` with the `.`. To clarify: it's currently getting either the prop name and uppercasing it or the `enumName` and treating those like the exported constant, then is appending `.${opt}` which was a key from the `options`. We just need to make it so the options **value**, not **key** can be used. 

Also: great work on this; super excited about it!

**Comment by tajo** [comment]
@EvanLovely Yep, that's how it works. Also, the [code generation](https://github.com/uber/react-view/blob/master/src/code-generator.ts#L47) needs to be updated so it outputs a string instead of the identifier.

**Comment by raunofreiberg** [comment]
Is anyone working on this? 🙂

**Comment by tajo** [comment]
@raunofreiberg not yet, do you want to work on it? happy to assist

**Comment by tajo** [comment]
I'm working on this now

**Comment by carmenbreezy** [comment]
Hello,

I came across this while implementing react-view as part of an internal design system. We also just use string literal unions instead of enums for our components. I noticed this has been closed but I was looking and don't happen to see an example where this is being used and I am still struggling a bit on how to work with react-view and string literal unions.

For example, in our design system we have a prop on our Button called `type` which has the string literal union of `"primary" | "secondary" | "tertiary" | "link"`. So using `useView` I'd expect to create something like this,
`  type: {
                value: 'secondary',
                defaultValue: 'secondary',
                options: ????,
                type: PropTypes.Enum,
                description: 'Type of button'
},`

What would I put in for `options`? Putting `["primary", "secondary", "tertiary", "link"]` just sets the values to their numeric equivalent, 0, 1, 2, 3.

Sorry I'm sure this is a really stupid question but I've been trying to figure it out and am just stumped.

Thank you!
